pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -525,14 +616,114 @@ pub mod pallet {
 			Ok((address, incoming_msg))
 		}
 
-		/// Sends the message to the `InboundMessageHandler`.
+		fn clear_storages_for_inbound_messages() {
+			let _ = InboundMessages::<T>::clear(u32::MAX, None);
+			let _ = InboundMessageProofCount::<T>::clear(u32::MAX, None);
+		}
+
+		fn get_expected_message_proof_count() -> u32 {
+			T::MultiRouterCount::get() - 1
```

Comment:
Again, we must use safe math as long as we cannot restrict `MultiRouterCount` to be at least one.

```suggestion
			T::MultiRouterCount::get().saturating_sub(1)
```

Reply:
Yes, most of the logic for inbound messages is super-dirty, will change all of these once I figure out proof processing accordingly.

Reply:
Makes absolute sense. Better have something working with some dirt than have a shining non-functional beauty 😅